### ISSUE TITLE
Implicit BuildFrom Is Not In Scope When Element Type Is Nothing

### ISSUE THREAD

**Issue by adamgfraser** [scala/bug#12104]
## reproduction steps

Using Scala 2.13.3,

```scala
import scala.concurrent._
import scala.concurrent.ExecutionContext.global
  
implicit val ec = global
  
Future.traverse(List(1))(_ => Future.failed(new NoSuchElementException))
// diverging implicit expansion for type scala.collection.BuildFrom[List[Int],B,List[B] starting with method Tuple9 in object Ordering
```

## problem

It looks like there is no implicit `BuildFrom` in scope when the inferred element type of the target collection is `Nothing`. There is some logic to this as we can't actually produce values of type `Nothing` or supply them to a `Builder` that requires them as inputs. But as the example above shows this causes significant problems. We can't traverse with `Future` when the result is statically known to be a failure.

I would think there should be a low priority instance of `BuildFrom[Any, Nothing, Collection[Nothing]]` in scope for standard types such as `Iterable` because we can always produce a `Builder[Nothing, Collection[Nothing]]` that can never accept elements but can build an empty collection of the target type.

Note that this is a regression from Scala 2.12, where this snippet would compile and produce a failed `Future`.

**Comment by dwijnand** [comment]
Notes: in 2.12 it infers `List.canBuildFrom[Nothing]`.

**Comment by SethTisue** [comment]
@julienrf @szeiger any thoughts on this? has this come up before?

**Comment by SethTisue** [comment]
Does binary compatibility allow us to add such a thing?

**Comment by julienrf** [comment]
It could, if there is an existing `BuildFrom` implicit definition that we can just “tweak” to also handle the `Nothing` case, and that would have the same erased signature as the existing definition.
Otherwise, it would have to go to scala-library-next? Does anyone want to have a look?

**Comment by som-snytt** [comment]
Fixed since 2.13.11.

Likely culprit https://github.com/scala/scala/pull/10209

without which the example does indeed fail again. What a spooky fix.
